File: backend/app/Models/KosRoomType.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

/**
 * Breakdown tipe kamar OPSIONAL per kos (mis. "Kamar AC" Rp1.8jt vs
 * "Kamar Standar" Rp1.2jt) -- MURNI TAMPILAN, tidak dipakai di alur
 * booking/ketersediaan (itu tetap dihitung di level Kos seperti biasa,
 * lihat catatan lengkap di migration create_kos_room_types_table).
 */
class KosRoomType extends Model
{
    use HasFactory;

    protected $fillable = ['kos_id', 'name', 'price', 'total_rooms'];

    public function kos()
    {
        return $this->belongsTo(Kos::class);
    }
}

File: frontend/lib/models/kos_room_type.dart
/// Breakdown tipe kamar OPSIONAL per kos (mis. "Kamar AC" Rp1.8jt vs
/// "Kamar Standar" Rp1.2jt) -- MURNI tampilan, tidak dipakai di alur
/// booking (penyewa tetap mengajukan booking ke kos secara umum, bukan
/// ke tipe kamar spesifik). Lihat backend KosRoomType untuk kenapa.
class KosRoomType {
  final int id;
  final String name;
  final int price;
  final int totalRooms;

  KosRoomType({
    required this.id,
    required this.name,
    required this.price,
    required this.totalRooms,
  });

  factory KosRoomType.fromJson(Map<String, dynamic> json) {
    return KosRoomType(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      price: json['price'] ?? 0,
      totalRooms: json['total_rooms'] ?? 0,
    );
  }
}
